python/pyiceberg/utils/parsing.py

```diff
@@ -32,4 +34,4 @@ def match(self, str_repr: str) -> int:
         matches = self.regex.search(str_repr)
         if matches:
             return int(matches.group(1))
-        raise ValueError(f"Could not match {str_repr}, expected format {self.prefix}[22]")
+        raise ValidationError(f"Could not match {str_repr}, expected format {self.prefix}[22]")
```

Comment:
Should `22` be hard-coded here?

Reply:
It is an example, it can be arbitrary. The `ParseNumberFromBrackets('ryan')` would accept anything like `ryan[22]`